### PR TITLE
Enable GitHub Code Scanning

### PR DESCRIPTION
**Pull request by HalcyonJAC**
**Author checklist**

- [ ] Include primary ticket number in title - e.g. "#123 New styling for widget" - and any additional tickets in description
- [ ] Fill in the details below and delete as appropriate
- [ ] Be proactive in getting your work approved 💪

---

## What's included?
A script that enables GitHub Code Scanning to run on a schedule.

## Who should test?
✅ Developers

## How to test?
Once the code has been merged, GitHub should start scanning the main develop every day and alerting us when it finds vulnerabilities in code, i.e. code that could allow remote code injection attacks, unsanitised inputs to be maniuplated, etc.

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

## Additional context
Include screen grabs, notes etc.

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

### CODE REVIEW

**Comment by lloback** [comment]
Please insert information for testing. Also there is no ticket on Zenhub for this change.

**Comment by HalcyonJAC** [comment]
> Please insert information for testing. Also there is no ticket on Zenhub for this change.

Done

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ develop ]
```
Comment (lloback) [inline .github/workflows/codeql-analysis.yml:16]: I think this is a great idea. 
We don't have the `develop` branch on almost all repositories. Isn't it now `main`?

**Review by lloback — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (HalcyonJAC) [reply .github/workflows/codeql-analysis.yml]: This has been changed from `develop` to `main`

**Review by lloback — APPROVED** [review]
✔︎✔︎✔︎

**Review by tomlovesgithub — APPROVED** [review]
Cool!

**Review by KateMJAC — APPROVED** [review]
Approved on the basis that both developer teirs approved.

**Pull request merged**